packages/slice-machine/lib/env/npmApi.ts

```diff
@@ -0,0 +1,58 @@
+import npmFetch from "npm-registry-fetch";
+import semver from "semver";
+import type { UpdateVersionInfo } from "../models/common/Environment";
+
+export async function fetchVersions(name: string): Promise<string[]> {
+  return npmFetch
+    .json(name)
+    .then((json) => {
+      const versions = (json.versions || {}) as Record<string, unknown>;
+      return Object.keys(versions);
+    })
+    .catch(() => []);
+}
+
+const highest = (
```

Comment:
the name is not very intuitive.
Since we're searching on an array, I would call it `findHighestUpdate` or something like that.

When a function has the name find, it implies that it's return can be undefined.